CS4375_AI_Techniques/practical_assignment_bayes.py

```diff
@@ -21,112 +21,72 @@
 from typing import List
 import unittest
 
+
 class Bayes:
-    def __init__(self, hypotheses: List[str], priors: List[float], observations: List[str], likelihood_array: List[List[float]]) -> None:
-        
-        self.hypotheses = hypotheses # List of hypothesis
+    def __init__(self, hypotheses: List[str], priors: List[float], observations: List[str],
+                 likelihood_array: List[List[float]]) -> None:
+
+        self.hypotheses = hypotheses  # List of hypothesis
         self.priors = priors  # List of prior probabilities for each hypothesis
         self.observations = observations  # List of possible observations
-        self.likelihood_array = likelihood_array # Likelihood array
-    
+        self.likelihood_array = likelihood_array  # Likelihood array
+
     def likelihood(self, observation: str, hypothesis: str) -> float:
         # From Likelihood Table get the index of observation and Hypothesis
-        hypothesis_index = self.hypotheses.index(hypothesis);
-        observation_index = self.observations.index(observation);
-        
-        return self.likelihood_array[hypothesis_index][observation_index];
-    
+        hypothesis_index = self.hypotheses.index(hypothesis)
+        observation_index = self.observations.index(observation)
+
+        return self.likelihood_array[hypothesis_index][observation_index]
+
     def norm_constant(self, observation: str) -> float:
-        '''
-        
-        Given The Bayes Rule, Normalizaiton constant is sum of Likelihood and priors (Read bayes Theorm)
-        
-        '''
-        
+        """
+
+        Given The Bayes Rule, Normalization constant is sum of Likelihood and priors (Read bayes Theorem)
+
+        """
+
         # Enumerate over the priors and Likelihood
-        # perform sum over index => (prior[index] * likelihhod[i])
-        norm_constant_value = 0;
+        # perform sum over index => (prior[index] * likelihood[i])
+        norm_constant_value = 0
         for index, element in enumerate(self.hypotheses):
             norm_constant_value += self.priors[index] * self.likelihood(observation, element)
-        
-        return norm_constant_value;
-    
-    def single_posterior_update(self, observation, priors) -> List[float]:
-        norm_constant =  self.norm_constant(observation);
+
+        return norm_constant_value
+
+    def single_posterior_update(self, observation) -> List[float]:
+        norm_constant = self.norm_constant(observation)
         # Storage for the posterior probabilities for each hypothesis
-        posteriors = []
+        priors = []
```

Comment:
I think this should be named posteriors, since this class methods is supposed to return the array(float) of posterior probabilities i.e. P(hypothesis | observation), prior is just P(observation)